### DISCUSSION TITLE
Add parser field to ParseException

### DISCUSSION THREAD

**Discussion by vsajip** [javacc21/javacc21#218] [Ideas]
I'd like to add to `ParseException` the following:
```java
    private XXXParser parser;

    public XXXParser getParser() { return parser; }
```
So that high-level code that catches a `ParseException` has access to the parser that threw it. It would certainly help with debugging!

**Comment by revusky** [comment]
Sure, seems like a good idea. A couple of people had some more specialized suggestions about some methods to add to ParseException, but I told them that they could, after all, just use injection, as in:

      INJECT ParseException : etc etc.

I mean, one would think that it would eventually start being people's instinctive reaction that, anything they need in a generated class, they can inject. However, your idea, access to the parser, requires adjusting the constructors that are invoked when a ParseException is instantiated, so that requires mucking with the core templates a bit. 

But yeah, go right ahead.

**Comment by vsajip** [comment]
Now tracked in #220.
